pkg/agent/openflow/packetin.go

```diff
@@ -68,28 +93,25 @@ const (
 	PacketInQueueRate = 100
 )
 
-// RegisterPacketInHandler stores controller handler in a map of map with reason and name as keys.
-func (c *client) RegisterPacketInHandler(packetHandlerReason uint8, packetHandlerName string, packetInHandler interface{}) {
+// RegisterPacketInHandler stores controller handler in a map with category as keys.
+func (c *client) RegisterPacketInHandler(packetHandlerCategory uint8, packetInHandler interface{}) {
 	handler, ok := packetInHandler.(PacketInHandler)
 	if !ok {
 		klog.Errorf("Invalid controller to handle packetin.")
 		return
 	}
-	if c.packetInHandlers[packetHandlerReason] == nil {
-		c.packetInHandlers[packetHandlerReason] = map[string]PacketInHandler{}
-	}
-	c.packetInHandlers[packetHandlerReason][packetHandlerName] = handler
+	c.packetInHandlers[packetHandlerCategory] = handler
```

Comment:
Is it guaranteed that each category will only have one handler?

Reply:
For now, yes. And each packetIn is sent by one OVS flow. So IMO, one packetIn will only relate to one feature/handler.

Reply:
It sounds sending packetIn multiple times for a packet has more cost, but it's ok for now since we don't have that case yet so I'm fine with the change.